### PR TITLE
05_As a user I need to find easily announcement by filtering the list

### PR DESCRIPTION
**Pull request by Jean-Beru**
Issue #4

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Resources/views/Base/index.html.twig [hunk 1]
```diff
@@ -48,20 +46,18 @@
         <div class="filter">
             <span class="title"><img class="sprite-icon-paperclip" alt=">" src="{{ asset('bundles/sensiolabsjobboard/images/blank.png') }}"/> Type of contract </span>
             <ul>
-                <li class="active">
-                    <a href="#">All types of contracts</a>
+                <li {% if filters.contract == null %}class="active"{% endif %}>
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Resources/views/Base/index.html.twig]: Use `is null` instead

File: src/SensioLabs/JobBoardBundle/Controller/BaseController.php [hunk 2]
```diff
@@ -26,7 +26,16 @@ public function indexAction(Request $request)
             );
         }
 
-        return ['jobs' => $jobs];
+        return [
+            'jobs' => $jobs,
+            'filter_country' => $repo->getCountriesCount($request->query->all()),
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Controller/BaseController.php]: `nbCountries` => ?
`nbContracts`=> ?

File: src/SensioLabs/JobBoardBundle/Entity/JobRepository.php [hunk 3]
```diff
@@ -3,23 +3,83 @@
...
+
+    /**
+     * @param array $filters
+     *
+     * @return array
+     */
+    public function getCountriesCount($filters = [])
+    {
+        $qb = $this->createQueryBuilder('j')
+            ->select('c.country, COUNT(j.id) AS nb_jobs')
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: do you really need c.country field ?

File: src/SensioLabs/JobBoardBundle/Resources/assets/homepage.js [hunk 4]
```diff
@@ -13,7 +13,11 @@ $(function () {
             $.ajax({
                 type: "GET",
                 url: loader.attr('data-url'),
-                data: {'page' : page},
+                data: {
+                    'page' : page,
+                    'country' : loader.attr('data-country'),
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/assets/homepage.js]: `loader.attr('data-country')` => `loader.data('country')` same for others attributes

File: src/SensioLabs/JobBoardBundle/Resources/views/Base/index.html.twig [hunk 5]
```diff
@@ -27,18 +27,16 @@
         <div class="filter">
             <span class="title"><img class="sprite-icon-location" alt=">" src="{{ asset('bundles/sensiolabsjobboard/images/blank.png') }}"/>Location</span>
             <ul>
-                <li class="active">
-                    <a href="#">All countries</a>
+                <li {% if filters.country == null %}class="active"{% endif %}>
```
Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Resources/views/Base/index.html.twig]: `if filters.country  is null`

File: src/SensioLabs/JobBoardBundle/Entity/JobRepository.php (hunk 3 above)
Comment (Jean-Beru) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: Yes, I use it to make URL or show country name.
eg : https://github.com/poledev/hubert/blob/issue-4/src/SensioLabs/JobBoardBundle/Resources/views/Base/index.html.twig#L35

Comment (salahm) [inline src/SensioLabs/JobBoardBundle/Entity/JobRepository.php]: okay, then the method name is not explicit enough :(

**Comment by salahm** [comment]
:+1:

**Pull request merged**